v-next/hardhat-keystore/src/internal/types.ts

```diff
@@ -0,0 +1,34 @@
+export type Format = "hh-unencrypted-keystore";
+
+export interface UnencryptedKeystoreFile {
+  _format: Format;
+  version: number;
+  keys: {
+    [key: string]: string;
+  };
+}
+
+export interface Keystore {
+  listKeys(): Promise<string[]>;
+  hasKey(key: string): Promise<boolean>;
```

Comment:
I added `hasKey` to separate out the checking for a key from the reading, as checking does not imply decrypting.